services/meta/client.go

```diff
@@ -1185,6 +1155,21 @@ func (c *Client) retryUntilSnapshot(idx uint64) *Data {
 	}
 }
 
+func (c *Client) updateAuthCache() {
+	// copy cached user info for still-present users
+	newCache := make(map[string]authUser, len(c.authCache))
```

Comment:
Won't this end up allocating a new map everytime the snapshot is updated even if there are no changes to the auth cache?